### ISSUE TITLE
Viewer euler1d No Response in Windows

### ISSUE THREAD

**Issue by terrychan999** [solvcon/solvcon#334]
When running the **euler1d** with Viewer on Windows and pressing the start button, there is no response.
However, when piping the viewer output to echo, it works properly.
Additionally, it functions properly when commenting out the `self.log` related code in `modmesh/app/euler1d.py`
https://github.com/solvcon/modmesh/blob/master/modmesh/app/euler1d.py#L631

![image](https://github.com/solvcon/modmesh/assets/14991138/8feb64ef-93ac-4cee-9d23-182f9b098ede)

#### Reference Build Process:
https://hackmd.io/XhxU2XnXQnqT4Of37kshVg

#### Build Commands:
```pwsh
cmake -Dpybind11_DIR="$(pybind11-config --cmakedir)" -S . -B build
cmake --build build --config Release --target viewer
```
#### Execute Commands:
Non-functioning Viewer:
```pwsh
.\build\cpp\binary\viewer\Release\viewer.exe
```
Functioning Viewer:
```pwsh
.\build\cpp\binary\viewer\Release\viewer.exe | echo
```
#### System Information:
- **OS:** Windows 11 23H2 22631.3593
- **modmesh git commit:** 880ea79
- **Compiler:** Visual Studio 17.9
- **Python:** 3.11.9
- **Qt:** 6.7.0
- **PySide6 / Shiboken6:** 6.7.0
- **CLI:** Windows Terminal + pwsh 7.4.2

**Comment by yungyuc** [comment]
Thanks @terrychan999 for identifying the issue!

> When running the **euler1d** with Viewer on Windows and pressing the start button, there is no response. However, when piping the viewer output to echo, it works properly. Additionally, it functions properly when commenting out the `self.log` related code in `modmesh/app/euler1d.py` https://github.com/solvcon/modmesh/blob/master/modmesh/app/euler1d.py#L631

I wonder how does it behave by commenting out the following two lines https://github.com/solvcon/modmesh/blob/f9c053e28f4212b5ab2956016ee0b81c16503e92/modmesh/app/euler1d.py#L732

```cpp
        view.mgr.pycon.writeToHistory(msg)
        view.mgr.pycon.writeToHistory('\n')
```

`view.mgr.pycon` is a Qt object.

**Comment by terrychan999** [comment]
> I wonder how does it behave by commenting out the following two lines
> ```c++
>         view.mgr.pycon.writeToHistory(msg)
>         view.mgr.pycon.writeToHistory('\n')
> ```

I have tried commenting out those two lines, No Response still happens.

Contrastly, when I commenting out the following two lines, it works:
https://github.com/solvcon/modmesh/blob/f9c053e28f4212b5ab2956016ee0b81c16503e92/modmesh/app/euler1d.py#L730
```c++
        sys.stdout.write(msg)
        sys.stdout.write('\n')
```

<br><br>

I've found that in Windows, a pythonw (GUI) app's `sys.__stdout__` may behave differently from a console app's, which I think might be the cause of this issue.
ref: [https://docs.python.org/3.12/library/sys.html#sys.\_\_stdout\_\_](https://docs.python.org/3.12/library/sys.html#sys.__stdout__)
> **Note:** Under some conditions `stdin`, `stdout` and `stderr` as well as the original values `__stdin__`, `__stdout__` and `__stderr__` can be `None`. It is usually the case for Windows GUI apps that aren’t connected to a console and Python apps started with pythonw.

**Comment by tychuang1211** [comment]
I can confirm that `sys.stdout` is `None` when not piping the output to console on Windows.

Add a single line before writing to stdout makes the app function properly. Thanks @terrychan999!

```python
if sys.stdout != None:
    sys.stdout.write(msg)
    sys.stdout.write('\n')
```

**Comment by yungyuc** [comment]
The issue is fixed by PR #359 .  Thank you, @tychuang1211 !
